virttest/_wrappers.py

```diff
@@ -0,0 +1,47 @@
+""" The file contains some wrappers of common python base-package workflows
+    that can be repeated throughout the avocado-vt code.
+    Please, remember that it is a private module.
+    Please, use this module ONLY from other virttest/avocado-vt modules.
+"""
+
+import sys
+import importlib
+
+
+def import_module(name, directory=None):
+    """ Imports a module named <name> to the execution environment.
+        If the module is not in the PYTHONPATH or the local directory,
+        its directory can be determined by the <directory> argument.
+        :param name: Name of the module that is going to be imported
+        :type name: String
+        :param directory: Path of the directory in which the module is located.
+                          If None, it will find the module in the current dir
+                          and the PYTHONPATH.
+        :type directory: String or None
+        :returns: The imported module
+    """
+    if directory:
+        sys.path.insert(0, directory)
+    # Just in case the module source has been dropped after
+    # the interpreter execution had begun
+    importlib.invalidate_caches()
+    module = importlib.import_module(name)
+    if directory:
+        sys.path.pop(0)
+    return module
```

Comment:
Looks we have to make the entire routine thread-safe as well, otherwise it might result in getting wrong priority of paths while importing modules within more than one thread context.

Reply:
I've been thinking about this one more deeply. It's in fact needed to ensure that the path priority is preserved, so I'll be adding a lock in the file to be acquired/released when modifying the `sys.path` during the process of importing a module.

However, after thinking about this issue for a bit, I met another concern: the fact that other modules running in other threads could be also modifying and inserting elements in `sys.path[0]` as well. Those threads wouldn't be aware of our (yet hypothetical) `_wrappers.local_lock` used to modify `sys.path` safely inside the `virttest._wrappers` module. In that case, we could be removing an arbitrary value when calling `sys.path.pop(0)` instead of the one that we actually want at the end of the import_module routine.

I've been trying to find if there is any lock that can help us protect that variable, but haven't found nothing that is really accessible and used by other modules that could be accessing `sys.path`. Instead, it seems that the most reasonable way of avoiding that case, is by calling `sys.path.remove(directory)` although its time complexity won't be O(1).

What do you think?